### DISCUSSION TITLE
How to know which filter has been done on a blog?

### DISCUSSION THREAD

**Discussion by blanquerna** [datenstrom/community#840] [Ask a question] [answered]
In some blog pages I can do a search by author or tag with the format: [ author=<authorName> ] or [ tag=<tagName> ]
This returns a list of pages that meet the condition of being from <authorName> or having the <tagName> tag.

In the title of the page appears the generic title of the blog pages and the condition [ Blog: <tagName> ]
I would like it to indicate [ Author:<authorName> ] or [ Tag: <tagName> ]

The functions that I have found in the API do not allow us to know if the response is from the author or from the tag.
I can look at the url, but the author or tag parameters do not appear.

What api query could I do to get if it is author or tag or see full url with parameters?

**Comment by markseuffert** [comment, marked as answer]
Have a look at `isRequest()`. This API function is also used by the blog extension to check if a specific filter was requested. Keep in mind that multiple filters can be combined, at the moment the blog supports `tag`, `author` and `published`. A good starting point is the source code of the extension you're using and the [API for developers](https://datenstrom.se/yellow/help/api-for-developers). 

Here are some examples:

* `$this->yellow->page->isRequest("tag")`: Check if page request argument exists for tag
* `$this->yellow->page->isRequest("author")`: Check if page request argument exists for author
* `$this->yellow->page->isRequest("published")`: Check if page request argument exists for published
* `$this->yellow->toolbox->isLocationArguments()`: Check if there are location arguments in current HTTP request

**Comment by blanquerna** [comment]
Thank you very much for the information and examples.

When I checked the API I didn't identify the function and how to use it.

Now I have been able to change the generic header of the [Blog] entry for the search type [Tag] or [Author]

> if($this->yellow->page->isRequest("tag"))  $searchParam="Tag:";
    else if($this->yellow->page->isRequest("author")) $searchParam="Author:";
    else $searchParam=preg_split("/\s*:\s*/", $this->yellow->page->getHtml("titleContent"))[0];
    echo "\<h1>".$searchParam." ".preg_split("/\s*:\s*/", $this->yellow->page->getHtml("titleContent"))[1]."\</h1>";

Alternatively you can use the corresponding icons:

> if($this->yellow->page->isRequest("tag"))  $searchParam='\<svg class="icon" style="transform: scale(2.5);">\<use xlink:href="#tag">\</use>\</svg>';
else if($this->yellow->page->isRequest("author")) $searchParam='\<svg class="icon" style="transform: scale(2);">\<use xlink:href="#author">\</use>\</svg>';
else $searchParam=preg_split("/\s*:\s*/", $this->yellow->page->getHtml("titleContent"))[0];
echo "\<h1>".$searchParam." ".$searchParam=preg_split("/\s*:\s*/", $this->yellow->page->getHtml("titleContent"))[1]."\</h1>";

It is also possible to customize the author icons with the images of the authors.
Thank you for a fantastic setting and your attention.
